🔧 CI gap: run-all.sh (L1–L4) should gate PRs to dev, not just the manual release-gate

**Process gap.** Per-PR CI on `dev` runs **CodeQL only**. The real test suite — `tests/run-all.sh` (L1 lint + L2 unit + L3 integration + L4 workflow-sim, incl. dist-fresh, shellcheck, agent-line-budget, local-agent-primitives) — is the `release-gate.yml` workflow, triggered by **workflow_dispatch / tag**, not on PRs or pushes to dev.

**Consequence (observed):** the #295–#301 merge batch landed **three dev-red regressions** that CodeQL didn't catch, only surfaced when run-all was run manually during #302:
- #285 `unpackEmbedding` → `readFloatLE` on a `Uint8Array` broke **all semantic/hybrid search** (12 L3 tests). 
- #293 world-model bullet pushed `pr-reviewer.md` to 31 lines (agent-line-budget cap 30).
- #298 test had a bare `cd` (shellcheck SC2164).

All three are now fixed (in #302), but they rode into dev clean because nothing gates PRs with the full suite.

**Fix:** run `tests/run-all.sh` (L1–L4; L0 docker-smoke can stay release-only) as a **required check on PRs targeting dev** — either a `pull_request` trigger in a new lightweight workflow, or extend `release-gate.yml` to also fire on `pull_request: [dev]` with the L6 chain skipped. This makes "dev is always green" enforceable rather than honor-system.

Aligns with the existing doctrine: *CI-affecting changes must pass the suite before merge; dev must always be green.*

--- a/.github/workflows/release-gate.yml
+++ b/.github/workflows/release-gate.yml
@@ -4,6 +4,8 @@ on:
     tags:
       - 'v[0-9]+.[0-9]+.[0-9]+-rc.[0-9]+'
       - 'v[0-9]+.[0-9]+.[0-9]+'
+  pull_request:
+    branches: [dev]
   workflow_dispatch:
     inputs:
       skip_l6:
@@ -87,19 +89,19 @@ jobs:
         run: bash tests/run-all.sh
 
       - name: Install Claude Code CLI (required by L6 chain runner)
-        if: ${{ inputs.skip_l6 != true }}
+        if: ${{ inputs.skip_l6 != true && github.event_name != 'pull_request' }}
         run: |
           npm install -g @anthropic-ai/claude-code
           claude --version
 
       - name: Run L6 13-step chain
-        if: ${{ inputs.skip_l6 != true }}
+        if: ${{ inputs.skip_l6 != true && github.event_name != 'pull_request' }}
         env:
           L6_FROM_STEP: ${{ inputs.l6_from_step || '1' }}
         run: bash tests/dogfood/run-l6-chain.sh --from "$L6_FROM_STEP" --fresh
 
       - name: Upload L6 chain artifacts (on success or failure)
-        if: ${{ always() && inputs.skip_l6 != true }}
+        if: ${{ always() && inputs.skip_l6 != true && github.event_name != 'pull_request' }}
         uses: actions/upload-artifact@v4
         with:
           name: l6-chain-runs-${{ github.run_id }}
@@ -109,6 +111,8 @@ jobs:
 
   install-smoke:
     name: L0 docker install-smoke
+    # L0 docker smoke stays release-only; PR gate is L1–L4 (run-all.sh).
+    if: ${{ github.event_name != 'pull_request' }}
     runs-on: ubuntu-latest
     timeout-minutes: 30
     steps:

